src/index.js

```diff
@@ -24,77 +24,98 @@ function App() {
         const pickedDueDate = dueDateInput.current.value;
 
-        if(currentInput && pickedDueDate) {
-            setToDos(prevList => {
-                return [...prevList, { id:uuidv4(), name: currentInput, due: pickedDueDate, completed: false}]
-            })
+        if (currentInput && pickedDueDate){ // will implement validations in next step
+            const newTodoItem = {
+                id: uuidv4(),
+                name: currentInput,
+                due: pickedDueDate,
+                completed: false
+            };
+
+            setToDos([...todos, newTodoItem]);
+            toDoInput.current.value = null;
+            dueDateInput.current.value = null;
         }
-        toDoInput.current.value = null;
-        dueDateInput.current.value = null;
     }
 
-    function toggleCompleted(e) {
-        const currentTodos = [...todos];
-        const toToggleItem = currentTodos.find(eachCurrentTodo => eachCurrentTodo.id === e);
+    const toggleCompleted = (taskID) => {
+        const toToggleItem = todos.find(({ id }) => id === taskID);
         toToggleItem.completed = !toToggleItem.completed;
-        setToDos(currentTodos);
+        setToDos([...todos]);
     }
 
-    function handleClearItems() {
-        const currentTodos = [...todos];
-        const toRemainItems = currentTodos.filter(eachCurrentTodo => !eachCurrentTodo.completed);
+    const handleClearItems = () => {
+        const toRemainItems = todos.filter(({ completed }) => !completed);
         setToDos(toRemainItems);
     }
 
-    function sortBy(e) {
-        const colToSort = e.target.value;
-        const currentSort = sortConfig;
-        const currentTodos = [...todos];
+    const toggleSort = (columnToSort) => {
         let direction;
 
-        if(colToSort){
-            if(!currentSort[colToSort]){
-                direction = 'asc';
-            }
-            else {
-                direction = currentSort[colToSort] === 'asc'? 'desc' : 'asc'; 
-            }
-            currentSort[colToSort] = direction;
-
-            setSortConfig(currentSort);
-
-            if(currentTodos) {
-                if(colToSort === 'date') {
-                    currentTodos.sort((a,b) => {
-                        if(+(new Date(a.due)) < +(new Date(b.due))) return direction === 'asc'? -1 : 1;
-                        if(+(new Date(a.due)) > +(new Date(b.due))) return direction === 'asc'? 1 : -1;
-                        return 0;
-                    });
+        // 1. if exists -->> flip it
+        // 2. if dont exists --> destroy everything in it, create this key
+        if (sortConfig['columnKey'] === columnToSort){  
+            direction = sortConfig['dirToSort'] === 'asc' ? 'des' : 'asc';
+        } else {
+            direction = 'asc';
+        }
+
+        const newSortConfig = { 
+            columnKey: columnToSort,
+            dirToSort: direction
+         };
+
+        setSortConfig(newSortConfig);
+    }
```

Comment:
Awesome! 👍 Storing sort config as a normalized object really makes this function not only clean but VERY easy to read.